flow/connectors/utils/monitoring/monitoring.go

```diff
@@ -112,19 +115,29 @@ func UpdateEndTimeForCDCBatch(
 	return nil
 }
 
-func AddCDCBatchTablesForFlow(ctx context.Context, pool shared.CatalogPool, flowJobName string,
-	batchID int64, tableNameRowsMapping map[string]*model.RecordTypeCounts,
+func AddCDCBatchTablesForFlow(
+	ctx context.Context,
+	pool shared.CatalogPool,
+	flowJobName string,
+	batchID int64,
+	tableNameRowsMapping map[string]*model.RecordTypeCounts,
+	otelManager *otel_metrics.OtelManager,
 ) error {
 	insertBatchTablesTx, err := pool.Begin(ctx)
 	if err != nil {
 		return fmt.Errorf("error while beginning transaction for inserting statistics: %w", err)
 	}
 	defer shared.RollbackTx(insertBatchTablesTx, internal.LoggerFromCtx(ctx))
-
+	tableNameOperations := make(map[string]map[string]int32, len(tableNameRowsMapping))
```

Comment:
Insert/Update/Delete doesn't need to be map, can be `[3]struct{op string;count int32}`